server/sockets/list.js

```diff
@@ -348,20 +367,193 @@ const emitRemoveMemberOnLeaveCohort = socket =>
+                removedViewers,
+                dashboardClients,
+                socket,
+                ListActionTypes.DELETE_SUCCESS,
+                listId
+              );
+            }
+
+            emitListForMany(
+              viewersIds,
+              dashboardClients,
+              socket,
+              ListActionTypes.FETCH_META_DATA_SUCCESS,
+              doc
+            );
+          }
+
+          if (cohortClients.size > 0) {
+            if (type === ListType.LIMITED && removedViewers) {
+              emitForMany(
+                removedViewers,
+                cohortClients,
+                socket,
+                ListActionTypes.DELETE_SUCCESS,
+                listId
+              );
+            }
+
+            emitListForMany(
+              viewersIds,
+              cohortClients,
+              socket,
+              ListActionTypes.FETCH_META_DATA_SUCCESS,
+              doc
+            );
+          }
+        }
+      });
+  });
+
+const removeMember = (socket, dashboardClients, listClients, cohortClients) =>
+  socket.on(ListActionTypes.REMOVE_MEMBER_SUCCESS, data => {
+    const { listId, userId } = data;
+
+    // Broadcast to every user on the list
+    socket.broadcast
+      .to(`sack-${listId}`)
+      .emit(ListActionTypes.REMOVE_MEMBER_SUCCESS, data);
+
+    // Broadcast to that removed user if he is at dashboard view
+    if (dashboardClients.has(userId)) {
+      socket.broadcast
+        .to(dashboardClients.get(userId))
+        .emit(ListActionTypes.DELETE_SUCCESS, listId);
+    }
+
+    // Broadcast to that removed user if he is on the list view
+    if (listClients.has(userId)) {
+      List.findById(listId)
+        .populate('cohortId')
+        .lean()
+        .exec()
+        .then(doc => {
+          if (doc) {
+            const { cohortId: cohort } = doc;
+            const { _id: cohortId } = cohort;
+
+            if (cohort) {
+              socket.broadcast
+                .to(listClients.get(userId))
+                .emit(ListActionTypes.REMOVE_MEMBER_SUCCESS, {
+                  cohortId,
+                  listId
+                });
+            }
+          }
+        });
+    }
+
+    // Broadcast to that removed user if he is on the cohort view
+    if (cohortClients.has(userId)) {
+      socket.broadcast
+        .to(cohortClients.get(userId))
+        .emit(ListActionTypes.DELETE_SUCCESS, listId);
+    }
+  });
+
+const archiveList = (socket, dashboardClients, cohortClients) =>
+  socket.on(ListActionTypes.ARCHIVE_SUCCESS, data => {
+    const { listId } = data;
+
+    // Broadcast to clients on list view
+    socket.broadcast
+      .to(`sack-${listId}`)
+      .emit(ListActionTypes.ARCHIVE_SUCCESS, data);
+
+    List.findById(listId)
+      .populate('cohortId')
+      .lean()
+      .exec()
+      .then(doc => {
+        const { viewersIds } = doc;
+
+        viewersIds.forEach(viewerId => {
+          dashboardClients.forEach((socketId, clientId) => {
```

Comment:
don't iterate through the dashboardClinets and cohortClients - this map may contain a lot of users. The native map's methods `has()` and `get()` are faster.
For each list viewer, you will iterate through the all dashboardClinets and second time through all clients on every cohort view...